ch2-es.md

```diff
@@ -0,0 +1,191 @@
+```
+
+El mundo está lleno de código ajax exactamente así. Esta es la razón por la que ambos son equivalentes:
+
+```js
+// esta línea
+return ajaxCall(function(json){
+  return callback(json);
+});
+
+// es la misma que esta línea
+return ajaxCall(callback);
+
+// así que podemos refactorizar getServerStuff
+var getServerStuff = function(callback){
+  return ajaxCall(callback);
+};
+
+// ...lo que es equivalente a esto
+var getServerStuff = ajaxCall; // <-- mira mamá, sin ()
+```
+
+Y eso, amigos, es cómo se hace. Una vez más, luego veremos por qué soy tan insistente.
+
+```js
+var BlogController = (function() {
+  var index = function(posts) {
+    return Views.index(posts);
+  };
+
+  var show = function(post) {
+    return Views.show(post);
+  };
+
+  var create = function(attrs) {
+    return Db.create(attrs);
+  };
+
+  var update = function(post, attrs) {
+    return Db.update(post, attrs);
+  };
+
+  var destroy = function(post) {
+    return Db.destroy(post);
+  };
+
+  return {
+    index: index, show: show, create: create, update: update, destroy: destroy
+  };
+})();
+```
+
+Este controlador ridiculo es 99% hueco, Podríamos, ya sea, reescribirlo como:
+
+```js
+var BlogController = {
+  index: Views.index,
+  show: Views.show,
+  create: Db.create,
+  update: Db.update,
+  destroy: Db.destroy
+};
+```
+
+...o desecharlo por completo, ya que no hace nada que no sea agrupar `Views` y `Db`.
+
+## ¿Por qué favorecer a las funciones de primera clase?
+
+Ok, vayamos a las razones de por qué favorecer a las funciones de primera clase. Como vimos en los ejemplos `getServerStuff` y `BlogController`, es fácil agregar capas de indirección que no tienen ningún valor real y sólo incrementan la cantidad de código para mantener y buscar.
+
+Además, si una función que estamos innecesariamente envolviendo sí cambia, también debemos cambiar nuestra función wrapper.
+
+```js
+httpGet('/post/2', function(json){
+  return renderPost(json);
+});
+```
+
+Si `httpGet` fuese a cambiar para enviar un posible error (`err`), necesitariamos cambiar la función interna.
+
+```js
+// tenemos que ir a cada llamada a httpGet en la aplicación y pasar explicitamente err.
+httpGet('/post/2', function(json, err){
+  return renderPost(json, err);
+});
+```
+
+Si la hubiéramos escrito como una función de primera clase, mucho menos tendría que cambiar:
+
+```js
+// renderPost es llamado dentro de httpGet, con todos los argumentos que quiera
+httpGet('/post/2', renderPost);
+```
+
+Además de la eliminación de funciones innecesarias, debemos nombrar y referenciar argumentos. Los nombres son casi un problema. Tenemos nombres inapropiados, especialmente, cuando el código base crece y los requerimientos cambian.
+
+Tener multiples nombres para el mismo concepto es una fuente común de confusión en proyectos. También existe el problema de código genérico. Por ejemplo, estas dos funciones hacen exactamente la misma cosa, pero una se siente infinitamente más general y reusable.
```

Comment:
'pero una es infinitamente más general y reusable.' en vez de 'pero una se siente infinitamente más general y reusable.'

En inglés, 'it feels like' o 'feels more' lo traduciría como 'parece como o más' en este contexto creo que es mejor no ponerlo ya que invita a confusión porque en español creo que no decimos una funcion se siente más reusable.

Que opináis?